container/docker/engine.py

```diff
@@ -213,6 +213,15 @@ def run_conductor(self, command, config, base_path, params, engine_name=None, vo
 
         if not volumes:
             volumes = {}
+
+        if params.get('with_volumes'):
+          for volume in params.get('with_volumes'):
+              volume_parts = volume.split(':')
+              volumes[volume_parts[0]] = {
+                  'bind': volume_parts[1],
+                  'mode': volume_parts[2] if len(volume_parts) > 2 else 'rw'
+              }
+
```

Comment:
According to [docker compose reference](https://docs.docker.com/compose/compose-file/#volumes), the short form of `volumes` may specify just the volume source. If so, then it's possible for volume_parts to have a length of 1, and for line 221 to throw an error.